### PR TITLE
add snippets

### PR DESCRIPTION
**Pull request by AtomSnippets**
Snippets are converted from Sublime Text (https://github.com/rust-lang/sublime-rust/tree/master/snippets) using the converter tool (https://github.com/james2doyle/sublime-to-atom-snippets)

### CODE REVIEW

**Comment by zargony** [comment]
@KalitaAlexey what do you think? Are these snippets useful to add or should we keep the current few ones and leave it to users to add more? I rarely use snippets at all, so I'm undecided about this.

File: snippets/rust.cson [hunk 1]
```diff
@@ -5,6 +5,23 @@
   'allow!':
     'prefix': 'allow!'
     'body': '#![allow(${1:lint})]'
+  "assert!(…)":
+    prefix: "assert"
+    body: "assert!(${1});"
+  "assert_eq!(…, …)":
+    prefix: "assert_eq"
+    body: "assert_eq!(${1}, ${2});"
+  undefined:
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: What does it mean?

File: snippets/rust.cson [hunk 2]
```diff
@@ -14,13 +31,43 @@
...
+     }
+   '''
   'enum':
     'prefix': 'enum'
     'body': '''
       enum ${1:TypeName} {
       \t$2
       }
     '''
+  "Err(…)":
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I don't like it.

File: snippets/rust.cson [hunk 3]
```diff
@@ -14,13 +31,43 @@
...
+     }
+   '''
+  "extern \"C\" { … }":
+    prefix: "extern-mod"
+    body: '''
+     extern "C" {
+       ${2:// add code here}
+     }
+   '''
+  "fmt!(…)":
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I don't understand it.

File: snippets/rust.cson [hunk 4]
```diff
@@ -42,13 +89,27 @@
       \t$3
       }
     '''
+  "if let … = … { … }":
+    prefix: "if-let"
+    body: '''
+     if let ${1:Some(pat)} = ${2:expr} {
+       ${2:unimplemented!();}
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I think you meant `${3}` here.

File: snippets/rust.cson [hunk 5]
```diff
@@ -42,13 +89,27 @@
...
+   '''
   'if':
     'prefix': 'if'
     'body': '''
       if ${1:expression} {
       \t$2
       }
     '''
+  "impl … for … { … }":
+    prefix: "impl-trait"
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: When `impl Trait` feature will be implemented, the name will conflict with the feature.
I'd suggest `impl-trait-for`.

File: snippets/rust.cson [hunk 6]
```diff
@@ -87,15 +155,37 @@
...
       }
     '''
+  "mod … { … }":
+    prefix: "mod"
+    body: '''
+     mod ${1:name} {
+       ${2:// add code here}
+     } 
+   '''
+  "Ok(…)":
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I don't like it.

File: snippets/rust.cson [hunk 7]
```diff
@@ -87,15 +155,37 @@
...
+    prefix: "Ok"
+    body: "Ok(${1:result})"
+  "panic!(…)":
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I don't understand it.

File: snippets/rust.cson [hunk 8]
```diff
@@ -87,15 +155,37 @@
...
+    prefix: "panic"
+    body: "panic!(\"${1:\\{${2::?}\\}}\"${1/([^\\{])*(\\{.*\\})?.*/(?2:, :\\);)/}$3${1/([^\\{])*(\\{.*\\})?.*/(?2:\\);)/}"
   'print':
     'prefix': 'print'
     'body': 'print!("${1:{${2::?}\\}}", ${3});'
   'println':
     'prefix': 'println'
     'body': 'println!("${1:{${2::?}\\}}", ${3});'
+  "Some(…)":
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I don't like it.

File: snippets/rust.cson [hunk 9]
```diff
@@ -140,6 +230,13 @@
   'warn!':
     'prefix': 'warn!'
     'body': '#![warn(${1:lint})]'
+  "while let … = … { … }":
+    prefix: "while-let"
+    body: '''
+     while let ${1:Some(pat)} = ${2:expr} {
+       ${2:unimplemented!();}
```
Comment (KalitaAlexey) [inline snippets/rust.cson]: I think you meant `${3}` here.

**Comment by zargony** [comment]
Closing because it seems unfinished and hasn't been active for a long time.

**Pull request closed without merging**